panic: Modify not implemented for node type <nil>

```
panic: Modify not implemented for node type <nil>
goroutine 82 [running]:
grol.io/grol/ast.Modify({0x0?, 0x0?}, 0x14000598ec8?)
	/Users/dl/dev/grol-io/grol/ast/modify.go:119 +0xe2c
grol.io/grol/ast.Modify({0x100a9e518?, 0x1400000fe00}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:44 +0x340
grol.io/grol/ast.Modify({0x100a9d518?, 0x1400040b6e0}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:12 +0x1518
grol.io/grol/ast.Modify({0x100a9e4c8?, 0x140002ed530}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:32 +0xec
grol.io/grol/ast.Modify({0x100a9d518?, 0x1400040b660}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:12 +0x1518
grol.io/grol/ast.Modify({0x100a9e540?, 0x140006119c0}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:52 +0x1124
grol.io/grol/ast.Modify({0x100a9e590?, 0x140002ec900}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:69 +0x13fc
grol.io/grol/ast.Modify({0x100a9d4c8?, 0x140002ec8d0}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:18 +0x438
grol.io/grol/ast.Modify({0x100a9d518?, 0x1400040b540}, 0x14000598ec8)
	/Users/dl/dev/grol-io/grol/ast/modify.go:12 +0x1518
grol.io/grol/eval.(*State).ExpandMacros(...)
	/Users/dl/dev/grol-io/grol/eval/macro_expension.go:77
grol.io/grol/eval.EvalString({0x100a8d4a0, 0x14000151830}, {0x140005ca000?, 0x100dbe110?}, 0x0)
	/Users/dl/dev/grol-io/grol/eval/eval_api.go:192 +0x2c4
grol.io/grol/repl.AutoLoad(0x14000151830, {0x0, 0x1, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, {0x0, ...}, ...})
	/Users/dl/dev/grol-io/grol/repl/repl.go:94 +0x220
grol.io/grol/repl.EvalStringWithOption({0x100a9f448, 0x100e3b920}, {0x0, 0x1, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	/Users/dl/dev/grol-io/grol/repl/repl.go:185 +0xd4
main.evalInput({0x140002ce080?, 0x3c400000000001?}, 0x14000599708)
	/Users/dl/dev/grol-io/grol-discord-bot/bot.go:313 +0x544
```

Somehow not triggered through grol tests but only discord bot - need to add a test that uses the same

repl.EvalStringWithOption with same options

--- a/ast/modify.go
+++ b/ast/modify.go
@@ -41,7 +41,9 @@ func Modify(node Node, f func(Node) Node) Node { //nolint:funlen,gocyclo // yeah
 		return f(newNode)
 	case *ReturnStatement:
 		newNode := &ReturnStatement{Base: node.Base}
-		newNode.ReturnValue = Modify(node.ReturnValue, f)
+		if node.ReturnValue != nil {
+			newNode.ReturnValue = Modify(node.ReturnValue, f)
+		}
 		return f(newNode)
 	case *FunctionLiteral:
 		newNode := *node

--- a/ast/modify.go
+++ b/ast/modify.go
@@ -3,7 +3,7 @@ package ast
 import "fmt"
 
 // Note, this is somewhat similar to eval.go's eval... both are "apply"ing.
-func Modify(node Node, f func(Node) Node) Node { //nolint:funlen,gocyclo // yeah lots of types.
+func Modify(node Node, f func(Node) Node) Node { //nolint:funlen,gocyclo,gocognit // yeah lots of types.
 	// TODO: add err checks for _s.
 	switch node := node.(type) {
 	case *Statements:

--- a/repl/repl_test.go
+++ b/repl/repl_test.go
@@ -2,6 +2,7 @@ package repl_test
 
 import (
 	"context"
+	"os"
 	"testing"
 
 	"grol.io/grol/eval"
@@ -169,5 +170,19 @@ func TestPreInputHook(t *testing.T) {
 	res, errs, _ := repl.EvalStringWithOption(context.Background(), opts, inp)
 	if res != expected || len(errs) > 0 {
 		t.Errorf("EvalString() got %v\n---\n%s\n---want---\n%s\n---", errs, res, expected)
+	}
+}
+
+func TestLikeDiscordBot(t *testing.T) {
+	input := "1"
+	expected := "1\n"
+	cfg := repl.EvalStringOptions()
+	cfg.AutoLoad = true
+	// create a .gr file:
+	content := `x=()=>{return}`
+	os.WriteFile(".gr", []byte(content), 0644)
+	evalres, errs, _ := repl.EvalStringWithOption(context.Background(), cfg, input)
+	if evalres != expected {
+		t.Errorf("EvalString() got %v\n---\n%s\n---want---\n%s\n---", errs, evalres, expected)
 	}
 }

--- a/repl/repl_test.go
+++ b/repl/repl_test.go
@@ -180,7 +180,7 @@ func TestLikeDiscordBot(t *testing.T) {
 	cfg.AutoLoad = true
 	// create a .gr file:
 	content := `x=()=>{return}`
-	os.WriteFile(".gr", []byte(content), 0644)
+	os.WriteFile(".gr", []byte(content), 0o644)
 	evalres, errs, _ := repl.EvalStringWithOption(context.Background(), cfg, input)
 	if evalres != expected {
 		t.Errorf("EvalString() got %v\n---\n%s\n---want---\n%s\n---", errs, evalres, expected)

